src/vs/workbench/browser/parts/views/viewPane.ts

```diff
@@ -230,6 +235,29 @@ export abstract class ViewPane extends Pane implements IView {
 		this._register(this.menuActions.onDidChange(() => this.updateActions()));
 
 		this.viewWelcomeController = new ViewWelcomeController(this.id, contextKeyService);
+		if (options.makeActivationEvent) {
+			if (this.isBodyVisible()) {
+				this.fireActivationEvent();
+			} else {
+				const fireActivation = this.onDidChangeBodyVisibility(visible => {
+					if (visible) {
+						this.fireActivationEvent();
+						fireActivation.dispose();
+					}
+				});
+				this._register(fireActivation);
+			}
+		}
+	}
+
+	private activated: boolean = false;
+	private async fireActivationEvent(): Promise<void> {
+		if (!this.activated) {
+			const promise = this.progressService.withProgress({ location: this.id }, () => this.extensionService.activateByEvent(`onView:${this.id}`))
+				.then(() => timeout(2000));
+			this.activated = true;
+			return promise;
+		}
 	}
```

Comment:
@sbatten does this seem ok to you? This is similar to how we do activation events for extension provided tree views.